### PR TITLE
implement GetSystemInformation

### PR DESCRIPTION
**Pull request by DenverM80**

### CODE REVIEW

File: test/service_tests.cpp [hunk 1]
```diff
@@ -57,3 +61,21 @@ BOOST_AUTO_TEST_CASE( put_bucket) {
...
+    ds3_request* request = ds3_init_get_system_information();
+    ds3_get_system_information_response* response;
+
+    printf("-----Testing GET system_information-------\n");
+
+    ds3_error* error = ds3_get_system_information(client, request, &response);
+    BOOST_CHECK(error == NULL);
+
+    BOOST_CHECK(response->api_version != NULL);
+    BOOST_CHECK(response->serial_number != NULL);
```
Comment (rpmoore) [inline test/service_tests.cpp]: What about the build information?  I don't see you getting that and verifying that it was not `NULL`

**Comment by rpmoore** [comment]
Would you please paste in the valgrind output for this build?

File: test/service_tests.cpp (hunk 1 above)
Comment (DenverM80) [inline test/service_tests.cpp]: in the sample response in the self documentation those fields were empty, but a curl request against a live system does indeed show info in those fields so I'll add those asserts

**Comment by DenverM80** [comment]
Meant to include this; I did run it before the pull request:

==25876== LEAK SUMMARY:
==25876==    definitely lost: 0 bytes in 0 blocks
==25876==    indirectly lost: 0 bytes in 0 blocks
==25876==      possibly lost: 0 bytes in 0 blocks
==25876==    still reachable: 2,550 bytes in 16 blocks
==25876==         suppressed: 0 bytes in 0 blocks
==25876== 
==25876== For counts of detected and suppressed errors, rerun with: -v
==25876== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
make: *** [mem] Error 201
denverm@dm-lt:~/code/github/ds3_c_sdk/test$

File: test/service_tests.cpp (hunk 1 above)
Comment (rpmoore) [inline test/service_tests.cpp]: Thanks!

**Pull request merged**